apis/python/src/tiledbsoma/experiment_query/query.py

```diff
@@ -0,0 +1,483 @@
+import concurrent.futures
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    ContextManager,
+    Dict,
+    Optional,
+    Sequence,
+    Tuple,
+    Union,
+    cast,
+)
+
+import anndata
+import attrs
+import numpy as np
+import numpy.typing as npt
+import pandas as pd
+import pyarrow as pa
+from typing_extensions import Literal, TypedDict
+
+from .. import somacore  # to be replaced by somacore package, when available
+from ..dataframe import DataFrame as SOMADataFrame
+from ..sparse_nd_array import SparseNDArrayRead
+
+if TYPE_CHECKING:
+    from ..experiment import Experiment
+from ..sparse_nd_array import SparseNDArray as SOMASparseNDArray
+from .anndata import _make_anndata
+from .axis import AxisQuery
+from .eq_types import AxisColumnNames, ExperimentAxisQueryReadArrowResult
+
+AxisJoinIds = TypedDict(
```

Comment:
It looks like these are only used internally. Should they be `_`-prefixed? (Also, since they aren’t used for their dict-ness at all I would make them attrs classes but that’s up to you.)

Reply:
made this change.  it cleans up some areas, but results in a lot of getattr/setattr code where the axis was parameterized.  Mixed result, but I think probably worth doing the change.